<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>870</width>
    <height>537</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>GridCal client</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/Program icon/GridCal_icon.svg</normaloff>:/Program icon/GridCal_icon.svg</iconset>
  </property>
  <property name="iconSize">
   <size>
    <width>24</width>
    <height>24</height>
   </size>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="GridTab">
       <attribute name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/Icons/white_icons/grid_icon.svg</normaloff>:/Icons/white_icons/grid_icon.svg</iconset>
       </attribute>
       <attribute name="title">
        <string>Grid data</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QSplitter" name="splitter_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QSplitter" name="splitter">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QListView" name="items_listView"/>
             </item>
             <item>
              <widget class="QFrame" name="frame_4">
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="refresh_items_pushButton">
                  <property name="text">
                   <string>Refresh list</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QFrame" name="frame_2">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="MatplotlibWidget" name="resultsPlot" native="true"/>
             </item>
             <item>
              <widget class="QComboBox" name="magnitude_comboBox"/>
             </item>
            </layout>
           </widget>
          </widget>
          <widget class="QFrame" name="frame_3">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QFrame" name="frame_5">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="1">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>val</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QDoubleSpinBox" name="max_val_doubleSpinBox">
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <double>10000.000000000000000</double>
                 </property>
                 <property name="value">
                  <double>10.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Min</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QDoubleSpinBox" name="min_val_doubleSpinBox">
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <double>1000.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSlider" name="value_horizontalSlider">
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Max</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QPushButton" name="send_pushButton">
                 <property name="text">
                  <string>Send</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="SettingsTab">
       <attribute name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/Icons/white_icons/gear.svg</normaloff>:/Icons/white_icons/gear.svg</iconset>
       </attribute>
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>70</y>
          <width>67</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>URL</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="url_lineEdit">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>70</y>
          <width>441</width>
          <height>25</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="status_label">
      <property name="text">
       <string>Status</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionLine">
   <property name="text">
    <string>Add Line</string>
   </property>
   <property name="toolTip">
    <string>Line</string>
   </property>
   <property name="shortcut">
    <string>Alt+L</string>
   </property>
  </action>
  <action name="actionTransformer">
   <property name="text">
    <string>Add Transformer</string>
   </property>
   <property name="shortcut">
    <string>Alt+T</string>
   </property>
  </action>
  <action name="actionLine_type">
   <property name="text">
    <string>Line type</string>
   </property>
   <property name="shortcut">
    <string>Alt+L, Alt+T</string>
   </property>
  </action>
  <action name="actionTransformer_type">
   <property name="text">
    <string>Transformer type</string>
   </property>
   <property name="shortcut">
    <string>Alt+T, Alt+T</string>
   </property>
  </action>
  <action name="actionOpen_file">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/loadc.svg</normaloff>:/Icons/white_icons/loadc.svg</iconset>
   </property>
   <property name="text">
    <string>Open file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/savec.svg</normaloff>:/Icons/white_icons/savec.svg</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="text">
    <string>Import</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/save.svg</normaloff>:/Icons/white_icons/save.svg</iconset>
   </property>
   <property name="text">
    <string>Export</string>
   </property>
  </action>
  <action name="actionNew_project">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/new2c.svg</normaloff>:/Icons/white_icons/new2c.svg</iconset>
   </property>
   <property name="text">
    <string>New project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionColor_by_type">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Color by type</string>
   </property>
  </action>
  <action name="actionColor_by_results">
   <property name="text">
    <string>Color by results</string>
   </property>
  </action>
  <action name="actionPower_flow">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/pf.svg</normaloff>:/Icons/white_icons/pf.svg</iconset>
   </property>
   <property name="text">
    <string>Power Flow</string>
   </property>
  </action>
  <action name="actionPower_Flow_Time_series">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/pf_ts.svg</normaloff>:/Icons/white_icons/pf_ts.svg</iconset>
   </property>
   <property name="text">
    <string>Power Flow: Time series</string>
   </property>
   <property name="toolTip">
    <string>Power flow time series</string>
   </property>
  </action>
  <action name="actionBlackout">
   <property name="text">
    <string>Blackout</string>
   </property>
  </action>
  <action name="actionMove_node_Up">
   <property name="text">
    <string>Move node Up</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Up</string>
   </property>
  </action>
  <action name="actionMove_node_Down">
   <property name="text">
    <string>Move node Down</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Down</string>
   </property>
  </action>
  <action name="actionMove_node_Left">
   <property name="text">
    <string>Move node Left</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Left</string>
   </property>
  </action>
  <action name="actionMove_node_Right">
   <property name="text">
    <string>Move node Right</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Right</string>
   </property>
  </action>
  <action name="actionBus_voltages">
   <property name="text">
    <string>Bus voltages</string>
   </property>
  </action>
  <action name="actionBranches_current">
   <property name="text">
    <string>Branches current</string>
   </property>
  </action>
  <action name="actionBranches_loading">
   <property name="text">
    <string>Branches loading (%)</string>
   </property>
  </action>
  <action name="actionBlackout_2">
   <property name="text">
    <string>Blackout</string>
   </property>
  </action>
  <action name="actionBranches_losses">
   <property name="text">
    <string>Branches losses</string>
   </property>
  </action>
  <action name="actionBus_voltages_p_u">
   <property name="text">
    <string>Bus voltages (p.u.)</string>
   </property>
  </action>
  <action name="actionBigger_nodes">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/plus.svg</normaloff>:/Icons/white_icons/plus.svg</iconset>
   </property>
   <property name="text">
    <string>Expand</string>
   </property>
   <property name="toolTip">
    <string>Expand distances</string>
   </property>
  </action>
  <action name="actionSmaller_nodes">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/minus.svg</normaloff>:/Icons/white_icons/minus.svg</iconset>
   </property>
   <property name="text">
    <string>Shrink</string>
   </property>
   <property name="toolTip">
    <string>Shrink distances</string>
   </property>
  </action>
  <action name="actionPower_flow_Stochastic">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/stochastic_power_flow.svg</normaloff>:/Icons/white_icons/stochastic_power_flow.svg</iconset>
   </property>
   <property name="text">
    <string>Power flow: Stochastic</string>
   </property>
   <property name="toolTip">
    <string>Monte Carlo stochastic power flow</string>
   </property>
  </action>
  <action name="actionActivate_map">
   <property name="text">
    <string>Activate map</string>
   </property>
  </action>
  <action name="actionVoltage_p_u">
   <property name="text">
    <string>Voltage (p.u.)</string>
   </property>
  </action>
  <action name="actionCurrent_p_u">
   <property name="text">
    <string>Current (p.u.)</string>
   </property>
  </action>
  <action name="actionLoading">
   <property name="text">
    <string>Loading (%)</string>
   </property>
  </action>
  <action name="actionLosses">
   <property name="text">
    <string>Losses</string>
   </property>
  </action>
  <action name="actionVoltage_stability">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/continuation_power_flow.svg</normaloff>:/Icons/white_icons/continuation_power_flow.svg</iconset>
   </property>
   <property name="text">
    <string>Voltage stability</string>
   </property>
   <property name="toolTip">
    <string>Voltage collapse</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionCenter_view">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/resize.svg</normaloff>:/Icons/white_icons/resize.svg</iconset>
   </property>
   <property name="text">
    <string>center view</string>
   </property>
   <property name="toolTip">
    <string>Center view</string>
   </property>
  </action>
  <action name="actionShort_Circuit">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/short_circuit.svg</normaloff>:/Icons/white_icons/short_circuit.svg</iconset>
   </property>
   <property name="text">
    <string>Short Circuit</string>
   </property>
  </action>
  <action name="actionAutoatic_layout">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/automatic_layout.svg</normaloff>:/Icons/white_icons/automatic_layout.svg</iconset>
   </property>
   <property name="text">
    <string>Autoatic layout</string>
   </property>
   <property name="toolTip">
    <string>Automatic layout the of the grid</string>
   </property>
  </action>
  <action name="actionLatin_Hypercube_Sampling">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/latin_hypercube2.svg</normaloff>:/Icons/white_icons/latin_hypercube2.svg</iconset>
   </property>
   <property name="text">
    <string>Latin Hypercube Sampling</string>
   </property>
   <property name="toolTip">
    <string>Latin Hypercube stochastic power flow</string>
   </property>
  </action>
  <action name="actionBlackout_cascade">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/blackout.svg</normaloff>:/Icons/white_icons/blackout.svg</iconset>
   </property>
   <property name="text">
    <string>Blackout cascade</string>
   </property>
   <property name="toolTip">
    <string>Run a simulation or step by step blackout cascade</string>
   </property>
  </action>
  <action name="actionRun_complete_cascade">
   <property name="icon">
    <iconset>
     <normaloff>:/file/run_cascade.svg</normaloff>:/file/run_cascade.svg</iconset>
   </property>
   <property name="text">
    <string>run complete cascade</string>
   </property>
  </action>
  <action name="actionRun_cascade_step">
   <property name="icon">
    <iconset>
     <normaloff>:/file/run_cascade_step.svg</normaloff>:/file/run_cascade_step.svg</iconset>
   </property>
   <property name="text">
    <string>Run cascade step</string>
   </property>
  </action>
  <action name="actionCopy_cascade_state_to_main_grid">
   <property name="icon">
    <iconset>
     <normaloff>:/white/white_icons/copy.svg</normaloff>:/white/white_icons/copy.svg</iconset>
   </property>
   <property name="text">
    <string>Copy cascade state to main grid</string>
   </property>
  </action>
  <action name="actionOPF">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/dcopf.svg</normaloff>:/Icons/white_icons/dcopf.svg</iconset>
   </property>
   <property name="text">
    <string>OPF</string>
   </property>
   <property name="toolTip">
    <string>Optimal power flow</string>
   </property>
  </action>
  <action name="actionOPF_time_series">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/dcopf_ts.svg</normaloff>:/Icons/white_icons/dcopf_ts.svg</iconset>
   </property>
   <property name="text">
    <string>OPF time series</string>
   </property>
   <property name="toolTip">
    <string>Optimal power flow time series</string>
   </property>
  </action>
  <action name="actionDetect_transformers">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/detect_tr.svg</normaloff>:/Icons/white_icons/detect_tr.svg</iconset>
   </property>
   <property name="text">
    <string>q</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Detect transformers.&lt;/p&gt;&lt;p&gt;Use the nodes nominal voltage to determine which branches should be a transformer.&lt;/p&gt;&lt;p&gt;If a branch joins two nodes with different voltage levels, the branch should be a transformer.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="actionAuto_rate_branches">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/Icons/white_icons/rate_br.svg</normaloff>:/Icons/white_icons/rate_br.svg</iconset>
   </property>
   <property name="text">
    <string>Auto rate branches</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Automatic rating of the branches.&lt;/p&gt;&lt;p&gt;Use the branches calculated power to establish a rate, if the branch rate is unknown. A factor is available in the settings.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MatplotlibWidget</class>
   <extends>QWidget</extends>
   <header>.matplotlibwidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
